MusicSchool/Controllers/CategoryController.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using MusicSchool.Models;
+using MusicSchool.Responses;
+
+namespace MusicSchool.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public class CategoryController : ControllerBase
+{
+    private readonly MusicSchoolDBContext _context;
+
+    public CategoryController(MusicSchoolDBContext context)
+    {
+        _context = context;
+    }
+
+    [HttpGet]
+    public async Task<ActionResult<IEnumerable<CategoryResponse>>> GetCategory()
+    {
+        var categories = await _context.Category
+            .OrderBy(c => c.CategoryName)
+            .Select(c => new CategoryResponse(c.Id, c.CategoryName, null))
+            .ToListAsync();
+
+        return Ok(categories);
+    }
+
+    // GET: api/Category/5
+    [HttpGet("{id}")]
+    public async Task<ActionResult<CategoryResponse>> GetCategory(int id)
+    {
+        var category = await _context.Category
+                .Where(x => x.Id == id)
+                .Select(x => new CategoryResponse(x.Id, x.CategoryName, x.Instruments))
+                .FirstOrDefaultAsync();
```

Comment:
```suggestion
                .SingleOrDefaultAsync(x => x.Id == id);
```

Reply:
👍I needed the include to get this to work.